commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java

```diff
@@ -432,10 +435,9 @@ public static ScheduledThreadPoolExecutor newScheduledThreadPoolExecutor(int cor
     @SuppressWarnings("DangerousThreadPoolExecutorUsage")
     public static ScheduledThreadPoolExecutor newScheduledThreadPoolExecutor(int corePoolSize,
             ThreadFactory threadFactory, RejectedExecutionHandler handler) {
-        if (corePoolSize == 0) {
-            corePoolSize = 1;
-        }
-        ScheduledThreadPoolExecutor ret = new ScheduledThreadPoolExecutor(corePoolSize, threadFactory, handler) {
+        int positiveCorePoolSize = corePoolSize > 0 ? corePoolSize : 1;
+        ScheduledThreadPoolExecutor ret = new ScheduledThreadPoolExecutor(positiveCorePoolSize, threadFactory,
+                handler) {
```

Comment:
I think this does actually change behaviour (if someone passed in negative values before, the `ScheduledThreadPoolExecutor` constructor would throw, while now we'll silently set it to 1) - given how widely used this class is I'd prefer for us not to change this too much.

Reply:
Fair, I'll fix this.